pkg/test/cel/backendtlspolicy_test.go

```diff
@@ -25,29 +25,172 @@ import (
+				},
+			}, {
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+				SectionName: ptrTo(gatewayv1a2.SectionName("foo")),
+			}},
+		},
+		{
+			name:       "invalid because duplicate target refs with duplicate section names",
+			wantErrors: []string{"sectionName must be unique when targetRefs includes 2 or more references to the same target"},
+			targetRefs: []gatewayv1a2.LocalPolicyTargetReferenceWithSectionName{{
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+				SectionName: ptrTo(gatewayv1a2.SectionName("foo")),
+			}, {
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+				SectionName: ptrTo(gatewayv1a2.SectionName("foo")),
+			}},
+		},
+		{
+			name:       "valid single targetRef without sectionName",
+			wantErrors: []string{},
+			targetRefs: []gatewayv1a2.LocalPolicyTargetReferenceWithSectionName{{
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+			}},
+		},
+		{
+			name:       "valid single targetRef with sectionName",
+			wantErrors: []string{},
+			targetRefs: []gatewayv1a2.LocalPolicyTargetReferenceWithSectionName{{
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+				SectionName: ptrTo(gatewayv1a2.SectionName("foo")),
+			}},
+		},
+		{
+			name:       "valid because duplicate target refs with different section names",
+			wantErrors: []string{},
+			targetRefs: []gatewayv1a2.LocalPolicyTargetReferenceWithSectionName{{
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+				SectionName: ptrTo(gatewayv1a2.SectionName("foo")),
+			}, {
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
+					Name:  "example",
+				},
+				SectionName: ptrTo(gatewayv1a2.SectionName("bar")),
+			}},
+		},
+		{
+			name:       "valid because duplicate target refs with different names",
+			wantErrors: []string{},
+			targetRefs: []gatewayv1a2.LocalPolicyTargetReferenceWithSectionName{{
+				LocalPolicyTargetReference: gatewayv1a2.LocalPolicyTargetReference{
+					Group: gatewayv1a2.Group(corev1.GroupName),
+					Kind:  gatewayv1a2.Kind("Service"),
```

Comment:
Would be good to add some test cases where the group or kind were the only parts that were different.